comms/rabbitmq.go

```diff
@@ -39,28 +43,33 @@ type rabbitMQComms struct {
 }
 
 // AdvertiseAgent will place an agent advertisement message on the message queue
-func (rmq rabbitMQComms) AdvertiseAgent(me defs.AgentAdvert) {
+func (rmq rabbitMQComms) AdvertiseAgent(me defs.AgentAdvert) error {
 
-	queue_url := fmt.Sprintf(
-		"amqp://%s:%s@%s:%s/",
+	queue_url := fmt.Sprintf("amqp://%s:%s@%s:%s/",
 		rmq.config.AMQP.User,
 		rmq.config.AMQP.Password,
 		rmq.config.AMQP.Host,
 		rmq.config.AMQP.Port,
 	)
 
+	retries := 0
 	conn, err := amqp.Dial(queue_url)
-	if err != nil {
-		log.Error(err)
-		log.Error("Failed to connect to RabbitMQ")
-		os.Exit(1)
+	for err != nil {
+		if retries > connectRetry {
+			log.Error("Failed to connect to RabbitMQ")
+			return err
+		}
+
+		retries++
+		log.Warnf("Failure connecting to RabbitMQ - retry #%s", string(retries))
```

Comment:
You probably want the %d formatter here. `string(int)` will give you the ASCII/UTF-8 character associated with the number.